vcr/pe/presentation_definition.go

```diff
@@ -473,70 +473,81 @@ func getValueAtPath(path string, vcAsInterface interface{}) (interface{}, error)
 // Supported go value types: string, float64, int, bool and array.
 // 'null' values are not supported.
 // It returns an error on unsupported features or when the regex pattern fails.
-func matchFilter(filter Filter, value interface{}) (bool, error) {
+func matchFilter(filter Filter, value interface{}) (bool, interface{}, error) {
 	// first we check if it's an enum, so we can recursively call matchFilter for each value
 	if filter.Enum != nil {
 		for _, enum := range filter.Enum {
 			f := Filter{
 				Type:  "string",
 				Const: &enum,
 			}
-			match, _ := matchFilter(f, value)
+			match, result, _ := matchFilter(f, value)
 			if match {
-				return true, nil
+				return true, result, nil
 			}
 		}
-		return false, nil
+		return false, nil, nil
 	}
 
 	switch typedValue := value.(type) {
 	case string:
 		if filter.Type != "string" {
-			return false, nil
+			return false, nil, nil
 		}
 	case float64:
 		if filter.Type != "number" {
-			return false, nil
+			return false, nil, nil
 		}
 	case int:
 		if filter.Type != "number" {
-			return false, nil
+			return false, nil, nil
 		}
 	case bool:
 		if filter.Type != "boolean" {
-			return false, nil
+			return false, nil, nil
 		}
 	case []interface{}:
 		for _, v := range typedValue {
-			match, err := matchFilter(filter, v)
+			match, _, err := matchFilter(filter, v)
 			if err != nil {
-				return false, err
+				return false, nil, err
 			}
 			if match {
-				return true, nil
+				return true, value, nil
 			}
 		}
 	default:
 		// object not supported for now
-		return false, ErrUnsupportedFilter
+		return false, nil, ErrUnsupportedFilter
 	}
 
 	if filter.Const != nil {
 		if value != *filter.Const {
-			return false, nil
+			return false, nil, nil
 		}
 	}
 
 	if filter.Pattern != nil && filter.Type == "string" {
 		re, err := regexp2.Compile(*filter.Pattern, regexp2.ECMAScript)
 		if err != nil {
-			return false, err
+			return false, nil, err
+		}
+		match, err := re.FindStringMatch(value.(string))
+		if err != nil {
+			return false, nil, err
 		}
-		return re.MatchString(value.(string))
+		if match == nil {
+			return false, nil, nil
+		}
+		// If there's a capture group, take the first one (not being the whole string, which is the first in the list)
```

Comment:
I think this should fail if a filter contains more than 1 capture group. The PD is invalid?

Reply:
I considered this, decided not to, but on second thought it's better to be more strict and make it more lenient if actually required.